Assets/MRTK/Core/Providers/Hands/BaseHandVisualizer.cs

```diff
@@ -189,7 +189,7 @@ public void OnHandMeshUpdated(InputEventData<HandMeshInfo> eventData)
                 mesh.normals = eventData.InputData.normals;
                 lastHandMeshVertices = eventData.InputData.vertices;
 
-                if (newMesh || meshChanged)
+                if (eventData.InputData.uvs != null && eventData.InputData.uvs.Length > 0)
```

Comment:
My gut says that we shouldn't revert this, since changing mesh data can get pretty expensive. especially if the data hasn't changed, it's probably best to avoid the below code.

Looks like the hand sends an empty meshinfo to alert the visualizer to clear the mesh: https://github.com/microsoft/MixedRealityToolkit-Unity/blob/cdd703dd1173ce59a01642da2933133006a125b3/Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityArticulatedHandDefinition.cs#L101-L103

Maybe if we can improve the `meshChanged` detection (like detecting going from a "cleared" mesh to actual data again). I think the `!= null` check prevents that:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/cdd703dd1173ce59a01642da2933133006a125b3/Assets/MRTK/Core/Providers/Hands/BaseHandVisualizer.cs#L180-L182

Maybe if the old array is null and the new one isn't, that's a `meshChanged` situation?